src/procevent.c

```diff
@@ -0,0 +1,1266 @@
+      is_match = (regexec(&pl->process_regex_obj, buffer,
+                          PROCEVENT_REGEX_MATCHES, matches, 0) == 0
+                      ? 1
+                      : 0);
+    } else {
+      is_match = (strcmp(buffer, pl->process) == 0 ? 1 : 0);
+    }
+
+    if (is_match == 1) {
+      DEBUG("procevent plugin: process %d name match (pattern: %s) for %s", pid,
+            (pl->is_regex == 0 ? pl->process : pl->process_regex), buffer);
+
+      if (pl->is_regex == 1) {
+        // If this is a regex name, copy the actual process name into the object
+        // for cleaner log reporting
+
+        if (pl->process != NULL)
+          sfree(pl->process);
+        pl->process = strdup(buffer);
+        if (pl->process == NULL) {
+          char errbuf[1024];
+          ERROR("procevent plugin: strdup failed during process_check: %s",
+                sstrerror(errno, errbuf, sizeof(errbuf)));
+          pthread_mutex_unlock(&procevent_list_lock);
+          return NULL;
+        }
+      }
+
+      if (pl->pid == pid) {
+        // this is a match, and we've already stored the exact pid/name combo
+        match = pl;
+        break;
+      } else if (pl->pid == -1) {
+        // this is a match, and we've found a candidate processlist_t to store
+        // this new pid/name combo
+        pl->pid = pid;
+        match = pl;
+        break;
+      } else if (pl->pid != -1) {
+        // this is a match, but another instance of this process has already
+        // claimed this pid/name combo,
+        // so keep looking
+        match = pl;
+        continue;
+      }
+    }
+  }
+
+  if (match != NULL && match->pid != -1 && match->pid != pid) {
+    // if there was a match but the associated processlist_t object already
+    // contained a pid/name combo,
+    // then make a new one and add it to the linked list
+
+    DEBUG(
+        "procevent plugin: allocating new processlist_t object for PID %d (%s)",
+        pid, match->process);
+
+    processlist_t *pl2;
+    char *process;
+    char *process_regex;
+
+    pl2 = malloc(sizeof(*pl2));
+    if (pl2 == NULL) {
+      char errbuf[1024];
+      ERROR("procevent plugin: malloc failed during process_check: %s",
+            sstrerror(errno, errbuf, sizeof(errbuf)));
+      pthread_mutex_unlock(&procevent_list_lock);
+      return NULL;
+    }
+
+    process = strdup(match->process);
+    if (process == NULL) {
+      char errbuf[1024];
+      sfree(pl2);
+      ERROR("procevent plugin: strdup failed during process_check: %s",
+            sstrerror(errno, errbuf, sizeof(errbuf)));
+      pthread_mutex_unlock(&procevent_list_lock);
+      return NULL;
+    }
+
+    if (match->is_regex == 1) {
+      pl2->is_regex = 1;
+      status =
+          regcomp(&pl2->process_regex_obj, match->process_regex, REG_EXTENDED);
+
+      if (status != 0) {
+        ERROR("procevent plugin: invalid regular expression: %s",
+              match->process_regex);
+        return NULL;
```

Comment:
have you leaked p12 and process at this point?

Reply:
I accidentally introduced these when I added regex functionality a little while back.  This also goes for the leaks found below.  Thank you for spotting them.